KioskApplication/src/main/java/controller/MapController.java

```diff
@@ -329,10 +327,109 @@ public void changed(ObservableValue<? extends Boolean> observable, Boolean oldVa
                     drawEdgesOnMap(MapEntity.getInstance().getEdgesOnFloor(floorSelector.getValue()));
                 }
                 else {
-                    for (javafx.scene.Node n: edgeObjectList) {
-                        nodesEdgesPane.getChildren().remove(n);
+                    DatabaseEdgeObjectList.clear();
+                }
+            }
+        });
+
+        DatabaseNodeObjectList.addListener(new ListChangeListener<Node>() {
```

Comment:
This method is too monolithic and the class might be getting to that point as well. I think the listeners are good but I think we should break them out into their own functions within this class or maybe in a different class.

Reply:
I can look into cleaning up the code if you want to focus on features.